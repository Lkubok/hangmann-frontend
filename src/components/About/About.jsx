import React, { Component } from "react";
import "./About.scss";

export class About extends Component {
  render() {
    return (
      <div className="about-box">
        <div className="about-section">
          <h2>Hangmann game github repository</h2>
          <a href="https://github.com/Lkubok/hangmann-game" target="blank">
            github.com/Lkubok/hangmann-game
          </a>
        </div>
        <div className="about-section">
          <h2>API used in game</h2>
          <a href="https://hangmann-backend.herokuapp.com" target="blank">
            hangmann-backend.herokuapp.com
          </a>
        </div>
        <div className="about-section">
          <h2>Contact With author</h2>
          <a href="mailto:Lkubok@gmail.com">Lkubok@gmail.com</a>
        </div>
        <div className="about-section about-section-inline">
          <h2>Used technologies</h2>
          <a
            target="_blank"
            rel="noopener noreferrer"
            href="https://reactjs.org"
          >
            ReactJS
          </a>
          <a
            target="_blank"
            rel="noopener noreferrer"
            href="https://redux.js.org/"
          >
            ReduxJS
          </a>
          <a
            target="_blank"
            rel="noopener noreferrer"
            href="https://jaredpalmer.com/formik/"
          >
            Formik
          </a>
          <a
            target="_blank"
            rel="noopener noreferrer"
            href="https://sass-lang.com/"
          >
            Sass
          </a>
          <a
            target="_blank"
            rel="noopener noreferrer"
            href="https://www.highcharts.com/"
          >
            HighCharts
          </a>
        </div>
      </div>
    );
  }
}
export default About;
