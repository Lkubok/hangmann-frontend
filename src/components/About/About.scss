@import "../../scss/mixins.scss";
@import "../../scss/variables.scss";

.about-box {
  @include flex-cc-col;
  flex-basis: 100%;
  .about-section {
    @include flex-cc-col;
    width: 100%;
    padding: 20px 0px;
    transition: all 0.4s ease-in-out;
    border-radius: 20px;
    &:hover {
      background-color: $dark-color-hover;
    }
    h2 {
      flex-basis: 100%;
      text-align: center;
    }
    a {
      text-decoration: none;
      color: $primary-color;
      padding: 5px 20px;
      transition: all 0.4s ease-in-out;
      &:hover {
        color: $primary-color-hover;
      }
    }
    &-inline {
      @include flex-cc-row;
    }
  }
}
