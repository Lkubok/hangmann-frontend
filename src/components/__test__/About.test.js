import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { About } from "../About/About";
Enzyme.configure({ adapter: new Adapter() });

describe("About component", () => {
  beforeEach(() => {});
  it("Should match snapshot", () => {
    const wrapper = shallow(<About />);
    expect(wrapper).toMatchSnapshot();
  });
});
