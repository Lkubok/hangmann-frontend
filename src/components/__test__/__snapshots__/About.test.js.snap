// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component Should match snapshot 1`] = `
<div
  className="about-box"
>
  <div
    className="about-section"
  >
    <h2>
      Hangmann game github repository
    </h2>
    <a
      href="https://github.com/Lkubok/hangmann-game"
      target="blank"
    >
      github.com/Lkubok/hangmann-game
    </a>
  </div>
  <div
    className="about-section"
  >
    <h2>
      API used in game
    </h2>
    <a
      href="https://hangmann-backend.herokuapp.com"
      target="blank"
    >
      hangmann-backend.herokuapp.com
    </a>
  </div>
  <div
    className="about-section"
  >
    <h2>
      Contact With author
    </h2>
    <a
      href="mailto:Lkubok@gmail.com"
    >
      Lkubok@gmail.com
    </a>
  </div>
  <div
    className="about-section about-section-inline"
  >
    <h2>
      Used technologies
    </h2>
    <a
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      ReactJS
    </a>
    <a
      href="https://redux.js.org/"
      rel="noopener noreferrer"
      target="_blank"
    >
      ReduxJS
    </a>
    <a
      href="https://jaredpalmer.com/formik/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Formik
    </a>
    <a
      href="https://sass-lang.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Sass
    </a>
    <a
      href="https://www.highcharts.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      HighCharts
    </a>
  </div>
</div>
`;
